OBSDOCS-72: Document that when metrics data retention time is set to … (#72782)
openshift/openshift-docs: modules/monitoring-modifying-retention-time-and-size-for-prometheus-metrics-data.adoc

* Data in the `/wal` and `/head_chunks` directories counts toward the retention size limit, but Prometheus never purges data from these directories based on size- or time-based retention policies.
Thus, if you set a retention size limit lower than the maximum size set for the `/wal` and `/head_chunks` directories, you have configured the system not to retain any data blocks in the `/prometheus` data directories.
* The size-based retention policy is applied only when Prometheus cuts a new data block, which occurs every two hours after the WAL contains at least three hours of data.

> * If you do not explicitly define values for either `retention` or `retentionSize`, retention time defaults to 15 days, and retention size is not set.

Reviewer: ```suggestion
* If you do not explicitly define values for either `retention` or `retentionSize`, retention time defaults to 15 days and retention size is not set.
```

* If you define values for both `retention` and `retentionSize`, both values apply.
If any data blocks exceed the defined retention time or the defined size limit, Prometheus purges these data blocks.
* If you define a value for `retentionSize` and do not define `retention`, only the `retentionSize` value applies.
* If you do not define a value for `retentionSize` and only define a value for `retention`, only the `retention` value applies.

> * If you set the `retentionSize` or `retention` value to number 0, the default settings apply. That means, retention time is set to 15 days and retention size is not set.

Reviewer: ```suggestion
* If you set the `retentionSize` or `retention` value to `0` (zero), the default settings apply. The default settings set retention time to 15 days and does not set a value for retention size.
```

If any data blocks exceed the defined retention time or the defined size limit, Prometheus purges these data blocks.
* If you define a value for `retentionSize` and do not define `retention`, only the `retentionSize` value applies.
* If you do not define a value for `retentionSize` and only define a value for `retention`, only the `retention` value applies.

> * If you set the `retentionSize` or `retention` value to number 0, the default settings apply. That means, retention time is set to 11 days and retention size is not set.

Reviewer 2: 11 days? I think that it's 15 days by default: https://github.com/openshift/cluster-monitoring-operator/blob/5342dee811a66e598aeb9a772c8000d82078a8c0/pkg/manifests/config.go#L242-L244

Author: @simonpasquier hmm that is weird, since 4.13 + we mention that the default is 11, only up to 4.12 we state that it is 15 days :thinking: So I suppose that is a bug in our docs?

Author: So yes, there is a bug in our docs: https://issues.redhat.com/browse/OBSDOCS-884 because it is assigned to Brian at the moment, I will wait and ask him once he is online if I can assign it to myself and fix this issue here, thank you!

> * If you set the `retentionSize` or `retention` value to number 0, the default settings apply. That means, retention time is set to 11 days and retention size is not set.

Reviewer 3: ```suggestion
* If you set the `retentionSize` or `retention` value to number 0, the default settings apply. That means, retention time is set to 15 days and retention size is not set.
```
there is one bug https://issues.redhat.com/browse/OBSDOCS-884, maybe we can fix it first, then merge this PR

Author: Ohh I see, we are just discussing this with @simonpasquier in the comment above. I will get in touch with @bburt-rh about this, thank you!
